nomic/project.py

```diff
@@ -0,0 +1,1370 @@
+                json={'atlas_index_id': self.atlas_index_id,
+                      'queries': base64.b64encode(bytesio.getvalue()).decode('utf-8'),
+                      'k': k},
+            )
+            status = response.status_code
+            retries += 1
+
+        if retries == 10:
+            raise AssertionError('Could not get response from server')
+
+        return response.json()
+
+
+class AtlasProject(AtlasClass):
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        description: Optional[str] = 'A description for your map.',
+        unique_id_field: Optional[str] = None,
+        modality: Optional[str] = None,
+        organization_name: Optional[str] = None,
+        is_public: bool = True,
+        project_id=None,
+        reset_project_if_exists=False,
+        add_datums_if_exists=True,
+    ):
+
+        """
+        Creates or loads an Atlas project.
+        Atlas projects store data (text, embeddings, etc) that you can organize by building indices.
+        If the organization already contains a project with this name, it will be returned instead.
+
+        **Parameters:**
+
+        * **project_name** - The name of the project.
+        * **description** - A description for the project.
+        * **unique_id_field** - The field that uniquely identifies each datum. If a datum does not contain this field, it will be added and assigned a random unique ID.
+        * **modality** - The data modality of this project. Currently, Atlas supports either `text` or `embedding` modality projects.
+        * **organization_name** - The name of the organization to create this project under. You must be a member of the organization with appropriate permissions. If not specified, defaults to your user account's default organization.
+        * **is_public** - Should this project be publicly accessible for viewing (read only). If False, only members of your Nomic organization can view.
+        * **reset_project_if_exists** - If the requested project exists in your organization, will delete it and re-create it.
+        * **project_id** - An alternative way to retrieve a project is by passing the project_id directly. This only works if a project exists.
+        * **reset_project_if_exists** - If the specified project exists in your organization, reset it by deleting all of its data. This means your uploaded data will not be contextualized with existing data.
+        * **add_datums_if_exists** - If specifying an existing project and you want to add data to it, set this to true.
+        **Returns:** project_id on success.
+
+        """
+        assert name is not None or project_id is not None, "You must pass a name or project_id"
+
+        super().__init__()
+
+        if project_id is not None:
+            self.meta = self._get_project_by_id(project_id)
+            return
+
+
+        results = self._get_existing_project_by_name(project_name=name, organization_name=organization_name)
```

Comment:
ah so this is why you cant return an AtlasProject...
this feels kinda awk - maybe just rename the function? like its not really getting a project.

Reply:
OH its only used in 1 place - here! just put the logic in this class?